src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -116,6 +117,20 @@ bool WholeBodyDynamicsDevice::closeExternalWrenchesPorts()
     return true;
 }
 
+bool WholeBodyDynamicsDevice::closeFilteredFTPorts()
+{
+    for(unsigned int i = 0; i < outputFTPorts.size(); i++ )
+    {
+        if( outputFTPorts[i] )
+        {
+            outputFTPorts[i]->close();
+            outputFTPorts[i].reset();
+            outputFTPorts[i] = nullptr;
```

Comment:
unique_ptr does not have to be deleted, it will automatically do the cleanup. You could just close the ports.